src/lib.rs

```diff
@@ -585,13 +591,23 @@ fn readline_edit<H: Helper>(
                     // line as the user typed it after a newline.
                     s.refresh_line_with_msg(None)?;
                 }
-                // Only accept value if cursor is at the end of the buffer
-                if s.validate()? && (cmd == Cmd::AcceptLine || s.line.is_end_of_input()) {
-                    break;
-                } else {
-                    s.edit_insert('\n', 1)?;
+                let valid = s.validate()?;
+                let end = s.line.is_end_of_input();
+                match (cmd, valid, end) {
+                    | (Cmd::AcceptLine, _, _)
+                    | (Cmd::AcceptOrInsertLine { .. }, true, true)
+                    | (Cmd::AcceptOrInsertLine { accept_in_the_middle: true }, true, _)
+                    => {
+                        break;
+                    }
+                    | (Cmd::Newline, _, _)
```

Comment:
Unreachable ?

Reply:
Fixed. Thanks!

(this is actually why I'd like the top-level match to be exhaustive, as this error would have been spotted by the compiler)